File: spring/actuator/src/main/java/com/hyfly/actuator/entity/message/MessageView.java
package com.hyfly.actuator.entity.message;

import com.alibaba.fastjson2.annotation.JSONField;
import com.hyfly.actuator.entity.message.constants.MessageLevel;
import com.hyfly.actuator.entity.message.constants.MessageType;
import lombok.Data;

@Data
public class MessageView {

    /**
     * info, error
     * {@link MessageLevel}
     */
    @JSONField(name = "@level")
    private String level;

    @JSONField(name = "@message")
    private String message;

    @JSONField(name = "@module")
    private String module;

    @JSONField(name = "@timestamp")
    private String timestamp;

    /**
     * planned_change, change_summary
     * {@link MessageType}
     */
    @JSONField(name = "type")
    private String type;

    @JSONField(name = "changes")
    private ChangeSummary changeSummary;

    @JSONField(name = "diagnostic")
    private Diagnostic diagnostic;
}

File: terraform/src/main/kotlin/com/hyfly/tf/actuator/entity/message/MessageView.kt
package com.hyfly.tf.actuator.entity.message

import com.alibaba.fastjson2.annotation.JSONField

class MessageView {
    /**
     * info, error
     */
    @JSONField(name = "@level")
    var level: String? = null

    @JSONField(name = "@message")
    var message: String? = null

    @JSONField(name = "@module")
    var module: String? = null

    @JSONField(name = "@timestamp")
    var timestamp: String? = null

    /**
     * planned_change, change_summary
     */
    @JSONField(name = "type")
    var type: String? = null

    @JSONField(name = "changes")
    var changeSummary: ChangeSummary? = null

    @JSONField(name = "diagnostic")
    var diagnostic: Diagnostic? = null
}
